src/function/memoize/shallowSingle.js

```diff
@@ -0,0 +1,20 @@
+import memoizeWith from './with';
+import identity from '../identity';
+
+/**
+ * Memoize functions with single argument of non-primitive argument.
+ * Uses Map instance for cache, because it's keys can be non-primitive.
+ * Use it when shallow equality for cache lookup is enough.
+ * It suitable for functions with single argument of primitive type too.
+ *
+ * @param {Function} fn
+ * @returns {Function}
+ * @example
+ *     const addFlag = obj => ({...obj, flag: true });
+ *     const memoize = memoizeWeak(addFlag));
+ *     const test = {test: 1};
+ *     memoize(test); // from addFlag call
+ *     memoize({test: 1}); // from addFlag call
+ *     memoize(test); // from cache
+ */
+export default memoizeWith(() => new Map(), identity);
```

Comment:
it isn't a shallowSingle as it stores stores many entries and uses SameValueZero check not a shallowEqual

Reply:
ah, sorry, I understood why it called Single - it uses only one argument,
should we add one more memoize function to store result only of last call, how do you think?

Reply:
done

Reply:
seems like the name of file is a little confusing:
it called shallow but doenst perform shallowEqual check